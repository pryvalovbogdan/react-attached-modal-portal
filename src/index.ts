import AttachedModal from './components/AttachedModalPortal';

export { AttachedModal };
