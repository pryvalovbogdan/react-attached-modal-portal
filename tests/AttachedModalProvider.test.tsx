import { render } from '@testing-library/react';
import React from 'react';

import { AttachedModalContextProvider, useAttachedModalContext } from '../src';
describe('AttachedModalContextProvider', () => {
  it('renders without crashing', () => {
    const { baseElement } = render(
      <AttachedModalContextProvider>
        <div>Test Child</div>
      </AttachedModalContextProvider>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('Should contain Desktop text based on property', () => {
    const Component = () => {
      const { isMobile } = useAttachedModalContext();

      return isMobile ? 'Mobile' : 'Desktop';
    };

    const { baseElement, getAllByText } = render(
      <AttachedModalContextProvider>
        <Component />
      </AttachedModalContextProvider>,
    );

    expect(getAllByText('Desktop')).toHaveLength(1);
  });
});
