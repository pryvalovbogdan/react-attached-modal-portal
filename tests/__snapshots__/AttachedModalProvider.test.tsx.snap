// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttachedModalContextProvider renders without crashing 1`] = `
<body>
  <div>
    <div>
      Test Child
    </div>
  </div>
</body>
`;
